pxtsim/simdriver.ts

```diff
@@ -601,6 +608,24 @@ namespace pxsim {
             if (!this.listener) {
                 this.listener = (ev: MessageEvent) => {
                     if (this.hwdbg) return
+
+                    if (U.isLocalHost()) {
+                        // no-op
+                    } else {
+                        let expectedOrigins: string[] =  [this.getSimUrl(), window.location.origin]
```

Comment:
we are processing lots of messages when doing radio or serial messages, can we streamline this check? could this whole thing be condensed to a single hashmap lookup?

Reply:
Totally. I created `messageOriginExpected` back when I didn't know there was the `simUrl` variable. I can actually just create a set of the origins I expect in the SimulatorDriver constructor and check against that set when I receive messages. 

#### Note
It seems TypeScript doesn't support HashSets, then again this set will only create 3 items so the O(n) lookup cost is negligible when using a normal Set. This will be in my next commit.

Reply:
I now just check if the event origin is in the set of allowed origins.